role-based-access-system/src/routes/roles.route.js

```diff
@@ -0,0 +1,82 @@
+import { Router } from "express";
+const route = Router();
+import query from "../db/query.js";
+
+// get all permissions for a role.
+route.get("/:roleId/permission", async (req, res) => {
+    try {
+        const roleId = req.params.roleId;
+        const permission = await query(`SELECT p.name FROM permission p INNER JOIN role_permission rp ON rp.permission_id = p.id WHERE rp.role_id =${roleId}`);
+        if (permission.affectedRows === 0) {
+            return res.send({ statusCode: 404, message: "Data not found" });
+        }
+        return res.send({ statusCode: 200, message: "Permissions fetched successfully", data: permission });
+    } catch (error) {
+        return res.send({ statusCode: 500, message: "Internal Server Error" });
+    }
+});
+
+//add permission to role
+route.post('/:roleId/permission/:permissionId', async (req, res) => {
```

Comment:
In post request values should be part of body

Reply:
changed request values in body